cmd/oras/internal/option/descriptor.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package option
+
+import (
+	"encoding/json"
+	"fmt"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/pflag"
+)
+
+// Descriptor option struct.
+type Descriptor struct {
+	OutputDescriptor bool
+}
+
+// ApplyFlags applies flags to a command flag set.
+func (opts *Descriptor) ApplyFlags(fs *pflag.FlagSet) {
+	fs.BoolVarP(&opts.OutputDescriptor, "descriptor", "", false, "output the descriptor")
+}
+
+// Marshal returns the JSON encoding of descriptor.
+func (opts *Descriptor) Marshal(desc ocispec.Descriptor) ([]byte, error) {
+	b, err := json.Marshal(desc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal descriptor: %w", err)
+	}
+	return b, err
```

Comment:
I think we don't need this function unless we need to add more info in the returned error. The price of adding this function is that you need to write more unit tests.

Reply:
Discussed offline. `json.Marshal` only returns go routine recover related errors. We still need this function to return descriptor related error message.